data/transactions/logic/assembler.go

```diff
@@ -1265,13 +1286,99 @@ func (ops *OpStream) checkStack(args StackTypes, returns StackTypes, instruction
 	}
 }
 
+const subRets int = -1
+const nowhere int = -2
+const exiting int = -3
+const erroring int = -4
+
+//A BasicBlock is a structure through which control can only flow in through start and out through end
+type BasicBlock struct {
+	startIndex       int         //index into ops.pending.Bytes()
+	endIndex         int         //Note this is the beginning of the last op in the block, somewhat unnecessary but we'll keep it for now
+	expectedStackTop []StackType //What the block requires to be on top of the stack in order to complete
+	stackDelta       []StackType //What the block net pushes to the stack
+	sourceErrors     []int       //Sourcelines for each entry in expectedStackTop which is useful for listing sourcelines for errors
+	jumpTo           int         //Index into ops.blocks, where control can jump to out of the block, i.e. via a branch
+	flowTo           int         //Where control can flow to, i.e. the block immediately following a conditional branch
+	callSubs         bool        //Whether or not the block contains a callsub instruction, redundant for now but potentially useful later if blocks get combined
+}
+
+func (ops *OpStream) fixJumpsAndFlows() {
+	refsIndex := 0
```

Comment:
I think a variable with "pc" is better here too.  Index is so overused, that anytime you can be more specific, you should.

Reply:
Does refCount work? I'm not sure if pc really fits, but maybe I'm misunderstanding what pc implies b/c my (potentially wrong) understanding was that it is used only in the context of program bytes.